### ISSUE TITLE
[Bug]: unable to apply with aws_cognito_user_pool_client

### ISSUE THREAD

**Issue by borfig** [hashicorp/terraform-provider-aws#30257]
### Terraform Core Version

1.3.0

### AWS Provider Version

4.60.0

### Affected Resource(s)

- aws_cognito_user_pool_client

### Expected Behavior

The `terraform apply` should succeed with no changes.

### Actual Behavior

The `terraform apply` fails with an error.

### Relevant Error/Panic Output Snippet

```shell
╷
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to
│ aws_cognito_user_pool_client.this, provider
│ "provider[\"registry.terraform.io/hashicorp/aws\"]" produced an unexpected
│ new value: .allowed_oauth_scopes: was cty.SetValEmpty(cty.String), but now
│ null.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
```

### Terraform Configuration Files

```terraform
resource "aws_cognito_user_pool_client" "this" {
  name            = "<some-name>"
  user_pool_id    = "<some-user-pool-id>"

  allowed_oauth_flows                  = []
  allowed_oauth_flows_user_pool_client = false
  allowed_oauth_scopes                 = []
  callback_urls                        = []
  logout_urls                          = []
  explicit_auth_flows                  = ["ALLOW_ADMIN_USER_PASSWORD_AUTH", "ALLOW_REFRESH_TOKEN_AUTH"]
  read_attributes                      = []
  supported_identity_providers         = ["COGNITO"]
  prevent_user_existence_errors        = "ENABLED"
}
```

### Steps to Reproduce

1. Deploy a user pool client with an older version of the provider (version 4.59.0 will do)
2. Update the terraform lock to install AWS provider version 4.60.0
3. run `terraform apply` without changing anything in the resource

### Important Factoids

The issue does not reproduce in version 4.59.0

### Would you like to implement a fix?

None

**Comment by joshlang** [comment]
Similar bug:

```
│ Error: Provider produced inconsistent result after apply
--
348 | │
349 | │ When applying changes to aws_cognito_user_pool_client.client, provider
350 | │ "provider[\"registry.terraform.io/hashicorp/aws\"]" produced an unexpected
351 | │ new value: .token_validity_units: block count changed from 0 to 1.
352 | │
353 | │ This is a bug in the provider, which should be reported in the provider's
354 | │ own issue tracker.
```

**Comment by gdavison** [comment]
Closed by #30459
